src/ledger/insert_shred.zig

```diff
@@ -1742,6 +1814,7 @@ const ShredSource = enum {
     recovered,
 };
 
+// QUESTION: Is this guaranteed to not have gaps ie start_index..end_index always?
```

Comment:
Yes, this struct is intended to indicate that we've received every single shred in the range. The logic is in updateCompletedDataIndexes.